automation/automation/apigw_body_handler.go

```diff
@@ -0,0 +1,41 @@
+package automation
+
+import (
+	"context"
+	"fmt"
+	"io"
+)
+
+type (
+	apigwBodyHandler struct {
+		reg apigwBodyHandlerRegistry
+	}
+)
+
+func ApigwBodyHandler(reg queueHandlerRegistry) *apigwBodyHandler {
+	h := &apigwBodyHandler{
+		reg: reg,
+	}
+
+	h.register()
+	return h
+}
+
+func (h apigwBodyHandler) read(ctx context.Context, args *apigwBodyReadArgs) (res *apigwBodyReadResults, err error) {
+	res = &apigwBodyReadResults{}
+
+	if !args.hasRequest {
+		err = fmt.Errorf("could not read body, contents missing")
+		return
+	}
```

Comment:
Is this needed if the param is marked as required?

Reply:
Sure, workflow probably handles that, will remove